CMakeLists.txt

```diff
@@ -34,42 +34,59 @@ else()
 endif()
 
 enable_testing()
-# Unit tests
-FILE(GLOB TestSources RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} test/*_test.cpp)
-foreach(testSrc ${TestSources})
-        get_filename_component(testName ${testSrc} NAME_WE)
-        add_executable(${testName} test/main-test.cpp ${testSrc})
-        target_link_libraries(${testName} PUBLIC ${Boost_UNIT_TEST_FRAMEWORK_LIBRARY})
-	add_test(${testName} ${testName})
-endforeach(testSrc)
 
-# Tests that should compile 
-FILE(GLOB TestCompileSources RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} test-compile/compiles/*_test.cpp)
-foreach(testCompSrc ${TestCompileSources})
-        get_filename_component(testCompName ${testCompSrc} NAME_WE)
-        add_executable(${testCompName} ${testCompSrc})
-        set_target_properties(${testCompName} PROPERTIES
-                              EXCLUDE_FROM_ALL TRUE
-                              EXCLUDE_FROM_DEFAULT_BUILD TRUE)
-        target_include_directories(${testCompName} PUBLIC test-compile/compiles)
-        add_test(NAME ${testCompName}
-                 COMMAND ${CMAKE_COMMAND} --build . --target ${testCompName}
-                 WORKING_DIRECTORY ${CMAKE_BINARY_DIR})
-endforeach(testCompSrc)
+#run all tests for sequential and concurrent versions
+set(EXEC_TYPES "seq" "par")
```

Comment:
It seems like you running both test suites against against a version linked to multithread boost. 
It is fine for local development, but we should have a flag to pass to link the singlethread version and skip the concurrent tests. That way, we can setup 2 CI environments to run against different versions in travisci.